vital/types/local_cartesian.cxx

```diff
@@ -0,0 +1,524 @@
+    Sin_Lat_Cos_Orient = Sin_LocalCart_Origin_Lat * Cos_LocalCart_Orientation;
+
+    N0 = semiMajorAxis / sqrt(1 - Geocent_e2 * Sin_LocalCart_Origin_Lat * Sin_LocalCart_Origin_Lat);
+
+    val = (N0 + LocalCart_Origin_Height) * Cos_LocalCart_Origin_Lat;
+    u0 = val * Cos_LocalCart_Origin_Lon;
+    v0 = val * Sin_LocalCart_Origin_Lon;
+    w0 = ((N0 * (1 - Geocent_e2)) + LocalCart_Origin_Height) * Sin_LocalCart_Origin_Lat;
+  }
+
+  /**
+   * @brief Converts geodetic coordinates to local cartesian coordinates
+   *
+   * The function convertFromGeodetic converts geodetic coordinates
+   * (latitude, longitude, and height) to local cartesian coordinates (X, Y, Z),
+   * according to the WGS84 ellipsoid and local origin parameters.
+   *
+   * @param [in]     geodetic_coordinate : WGS84 longitude/latitude in degrees and hight in meters
+   * @param [in,out] cartesian_coordinate : The location in the local coordinate system, in meters
+   */
+  void convert_from_geodetic( vector_3d const& geodetic_coordinate, vector_3d& cartesian_coordinate ) const
+  {
+    double longitude = geodetic_coordinate.x() * DEG_TO_RAD;
+    double latitude = geodetic_coordinate.y() * DEG_TO_RAD;
+    double height = geodetic_coordinate.z();
+
+    double Rn;            /*  Earth radius at location  */
+    double Sin_Lat;       /*  sin(Latitude)  */
+    double Sin2_Lat;      /*  Square of sin(Latitude)  */
+    double Cos_Lat;       /*  cos(Latitude)  */
+
+    if (longitude > PI)
+    {
+      longitude -= (2 * PI);
+    }
+    Sin_Lat = sin(latitude);
+    Cos_Lat = cos(latitude);
+    Sin2_Lat = Sin_Lat * Sin_Lat;
+    Rn = semiMajorAxis / (sqrt(1.0e0 - Geocent_e2 * Sin2_Lat));
+    double X = (Rn + height) * Cos_Lat * cos(longitude);
+    double Y = (Rn + height) * Cos_Lat * sin(longitude);
+    double Z = ((Rn * (1 - Geocent_e2)) + height) * Sin_Lat;
+
+    vector_3d geocentric_coordinate;
+    geocentric_coordinate << X, Y, Z;
+
+    convert_from_geocentric(geocentric_coordinate, cartesian_coordinate);
+  }
+
+  /**
+   * @brief Converts local cartesian coordinates to geodetic coordinates
+   *
+   * The function convertToGeodetic converts local cartesian
+   * coordinates (X, Y, Z) to geodetic coordinates (latitude, longitude,
+   * and height), according to the WGS84 ellipsoid and local origin parameters.
+   *
+   * @param [in]     cartesian_coordinate : The location in the local coordinate system, in meters
+   * @param [in,out] geodetic_coordinate  : WGS84 longitude/latitude in degrees and hight, in meters
+   */
+  void convert_to_geodetic( vector_3d const& cartesian_coordinate, vector_3d& geodetic_coordinate ) const
+  {
+    vector_3d geocentric_coordinate;
+    convert_to_geocentric(cartesian_coordinate, geocentric_coordinate);
+
+    double X = geocentric_coordinate.x();
+    double Y = geocentric_coordinate.y();
+    double Z = geocentric_coordinate.z();
+    double latitude, longitude, height;
+
+    if (Iterative_Geocent_algorithm)
```

Comment:
What does this compile-time switch do?  Should we just remove the code that is not being used?  Or maybe make it a run time option?  It looks like there are two algorithm alternatives, do we know the pros and cons of each?

Reply:
Well, it looks like they released geotrans 3.8 this year, but I need certs to download it...
It looks like they added spherical coordinate support and fixed a but we may want to get (if I can get the source code)

https://earth-info.nga.mil/GandG/update/wgs84/apps/geotrans/current-version/docs/MSP_Geotrans_Release_Notes%203.8.pdf